pkg/cmd/server/server.go

```diff
@@ -246,6 +259,7 @@ func (s *server) run() error {
 	// separate object, and instead apply defaults to a clone.
 	config := originalConfig.DeepCopy()
 	applyConfigDefaults(config, s.logger)
+	s.applyConfigDefaults()
```

Comment:
I don't like having this func be separate from the previous function, esp. with the same names.  I think it makes sense to keep the new method you created with the server receiver, to pass it the *api.Config object, and have it do what it's already doing plus whatever's left in the old `applyConfigDefaults`, then we can get rid of the old func.